pyscal/pyscallist.py

```diff
@@ -254,10 +257,9 @@ def dump_family_2(self, filename: Optional[str] = None) -> str:
         if len(self) == 0:
             return ""
         if self.pyscaltype == SCALrecommendation:
-            logger.error(
+            raise TypeError(
```

Comment:
In each of these methods of `PyscalList` you have many many checks for `pyscaltype`. Why not do

`class _PyscalList` <- base behaviour, if any
`class SCALrecommendation(_PyscalList)`

It could greatly reduce complexity of all of the methods of `PyscalList`, where you could assume that `pyscaltype` is always one thing.

Of course, the assumption here is that `pyscaltype` is only one type at a time.

Reply:
This is left as issue #328